app/getPairNumberFromColor.js

```diff
@@ -0,0 +1,26 @@
+import { MAJOR_COLOR_NAMES, MINOR_COLOR_NAMES } from "./colorNames.constants.js";
+
+export function getPairNumberFromColor(pair) {
+  let majorColorIndex = -1;
+  for (let i = 0; i < MAJOR_COLOR_NAMES.length; i++) {
+    if (MAJOR_COLOR_NAMES[i] == pair.majorColor) {
+      majorColorIndex = i;
+      break;
+    }
+  }
+
+  let minorColorIndex = -1;
```

Comment:
See repeated pattern in major and minor color names - initialize to -1, then iterate and compare.
Try moving this functionality into a function and remove the repeated patterns by calling that function.